Add service layer boilerplate

dbrosio3 opened the pull request:
## Links:
<!--- At a minimum include a link to the Ticket it implements --->

* [Trello Card](https://trello.com/c/5QOv8lHL/2-2-as-a-user-i-should-be-able-to-sign-up-to-create-an-account)

## What & Why:
<!--- Describe the changes being made and why they're useful --->

Setup api client to start implementing the registration
* Created enviroment variables
* Configured axios client
* Linting fixes

<!--- Screenshots are expected for UI changes... right? ಠ_ಠ --->

N/A

## Risks, Mitigation & Rollback:
<!--- What risks exist for these new changes and what steps to mitigate them have been taken? --->
<!--- What steps are necessary to rollback this code safely? --->

- [x] Safe to Revert *check this box if this PR can be reverted without incident*

dbrosio3 committed:
add service layer boilerplate

dbrosio3 committed:
remove eslint warnings

dbrosio3 committed:
add private and public routing

dbrosio3 commented on .eslintrc.js:
```diff
@@ -52,13 +52,14 @@ module.exports = {
-        'import/no-named-as-default-member': 'off',
-        '@typescript-eslint/no-unused-vars': ['error'],
+        "import/no-named-as-default": 'off',
+        "@typescript-eslint/no-unused-vars": ["warn", { "varsIgnorePattern": "^_", "argsIgnorePattern": "^_" }],
```
The idea here is to ignore the unused vars rule for variable names `_likeThis`

elizabethlofredo replied:
curious, which case are you covering with this change? 🤔 I believe we should remain the error for all unused variables

dbrosio3 replied:
I think it's useful during development to declare a variable that maybe you don’t use right away and it still compiles with a warning.

The _ is to remove that warning on some specific cases:
- unused function arguments.
```
function fn(_first, second) { ... }
```
- for destructuring an array:
```
const [_head, tail] = someArray;
```
- [here](https://github.com/dbrosio3/blackmarket-frontend/blob/f7281ff20be69ed52c29dc2a10a06347c92c7166/src/features/register/Register.tsx#L12), for example, I plan on using the `response` but I don’t feel like defining the necessary types to pass it as an argument of `onRegister` yet. 

This last case is a bit of a stretch, I'll remove that variable.
(The warning was a bit annoying so I intentionally ignored it 🙈)

elizabethlofredo replied:
I believe the general rule should be to have an error and then disable the specific cases where you want to ignore it

elizabethlofredo commented on src/features/register/api/register.ts:
```diff
@@ -0,0 +1,37 @@
+
+export const registerWithEmailAndPassword = (
+  data: RegisterCredentialsDTO
+): Promise<RegistrationResponse> => {
+  return axios.post('/auth', data, {
```
minor comment, you can remove the return here

```suggestion
): Promise<RegistrationResponse> => axios.post('/auth', data, {
```

dbrosio3 committed:
add discussed suggestions

dbrosio3 committed:
run `git config core.ignorecase false`

was having issues with renamed directories. git didn't seem to track the changes where it was only an upper/lower case change

dbrosio3 committed:
fix naming missmatch with filesystem error

dbrosio3 committed:
add paths enum

marcio-b commented on src/lib/axios.ts:
```diff
@@ -0,0 +1,7 @@
+
+import { API_URL } from '@/config';
+
+export const axios = Axios.create({
```
Though this might be a valid export, I would suggest not exporting your instance with the same name as the module to avoid accidentally importing from the `axios` module itself (in case you have an auto-import extension for example).

marcio-b commented on src/providers/CartContext.tsx:
```diff
@@ -23,6 +23,6 @@ const CartProviderComponent: React.FC<JustChildrenProp> = ({ children }) => {
 
 const CartProvider = CartProviderComponent;
```
Is there a reason why you Created the provider component and then assign it to a different variable?

dbrosio3 replied:
It's just for export purposes. I don't like the name `<CartProviderComponent>` in the jsx

Now that you mention it, I could just directly name it `CartProvider`

marcio-b commented on src/providers/SessionContext.tsx:
```diff
@@ -39,7 +41,7 @@ const SessionProviderComponent: React.FC<JustChildrenProp> = ({ children }) => {
         login,
         logout,
-        register,
+        onRegister,
```
If one of the exported functions gets renamed... WDYT about renaming all of them to provide consistency?

dbrosio3 replied:
Makes sense. I was thinking of changing them with the auth implementation. Once I have a more solid understanding of how it all fits together. Maybe I change it now and end up changing them all again later.

flomosq replied:
I agree with Marcio's comment but if you're changing them with the auth implementation then it's fine by me

marcio-b reviewed (commented):
Looks good to me! I like the way you typed functions in general. I left a couple of small comments and/or questions.

dbrosio3 committed:
remove unnecesary provider renaming

dbrosio3 committed:
rename axios client

dbrosio3 committed:
fix git missmatch

flomosq reviewed (approved):
Looking good 🚀

dbrosio3 committed:
renamed auth routes

dbrosio3 committed:
add TODO and clarification comment

dbrosio3 committed:
remove empty lines in AppRoutes so it looks cleaner

dbrosio3 committed:
Merge pull request #5 from dbrosio3/feature/private-and-public-routing
